File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> roman_values = new HashMap<>();
        roman_values.put('I', 1);
        roman_values.put('V', 5);
        roman_values.put('X', 10);
        roman_values.put('L', 50);
        roman_values.put('C', 100);
        roman_values.put('D', 500);
        roman_values.put('M', 1000);

        int count = 0;
        int curr = 0;
        int last = 0;

        for (int i = 0; i < s.length(); i++) {
            curr = roman_values.get(s.charAt(i));
            count += curr;
            if (curr > last) {
                count -= 2*last;
            }
            last = curr;
        }

        return count;
    }
}

File: 13-roman-to-integer/roman-to-integer.c
int romanCharToInt(char c) {
    switch (c) {
        case 'I' : return 1;
        case 'V' : return 5;
        case 'X' : return 10;
        case 'L' : return 50;
        case 'C' : return 100;
        case 'D' : return 500;
        case 'M' : return 1000;
        default : return 0;
    }
}

int romanToInt(char* s) {
    int count = 0;
    int curr = 0;
    int last = 0;
    while (*s != '\0') {
        curr = romanCharToInt(*s);
        count += curr;
        if (curr > last) {
            count -= 2 * last;
        }
        last = curr;
        s += 1;
    }
    return count;
}